include/deal.II/base/exceptions.h

```diff
@@ -298,13 +299,34 @@ namespace deal_II_exceptions
           dealii::deal_II_exceptions::internals::abort(e);
           break;
         case abort_nothrow_on_exception:
-          dealii::deal_II_exceptions::internals::abort(e, /*nothrow =*/ true);
+          // This case should not be handled here.
+          issue_error_nothrow(handling, file, line, function, cond, exc_name, e);
           break;
         case throw_on_exception:
           throw e;
         }
     }
 
+    /**
+     * Exception generation mechanism in case we must not throw.
+     *
+     * @ref ExceptionBase
+     */
+    template <class ExceptionType>
+    void issue_error_nothrow (ExceptionHandling,
```

Comment:
you could make the last argument of type `ExceptionBase`. In that case, you don't need the template, can move the implementation into the .cc file, and remove the `#include <iostream>` again.